lib/helpers/discriminator/mergeDiscriminatorSchema.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+const schemaMerge = require('../schema/merge');
+const { trustedSymbol } = require('../../helpers/query/trusted');
+const specialProperties = require('../../helpers/specialProperties');
+const isBsonType = require('../../helpers/isBsonType');
+const ObjectId = require('../../types/objectid');
+const isObject = require('../../helpers/isObject');
+/**
+ * Merges `from` into `to` without overwriting existing properties.
+ *
+ * @param {Object} to
+ * @param {Object} from
+ * @param {Object} [options]
+ * @param {String} [path]
+ * @api private
+ */
+
+module.exports = function mergeDiscriminatorSchema(to, from, options, path) {
+  options = options || {};
+
+  const keys = Object.keys(from);
+  let i = 0;
+  const len = keys.length;
+  let key;
+
+  if (from[trustedSymbol]) {
+    to[trustedSymbol] = from[trustedSymbol];
+  }
+
+  path = path || '';
+  const omitNested = options.omitNested || {};
+
+  while (i < len) {
+    key = keys[i++];
+    if (options.omit && options.omit[key]) {
+      continue;
+    }
+    if (omitNested[path]) {
```

Comment:
Get rid of `omitNested` and replace this check with `if (path === 'tree' && from != null && from.instanceOfSchema)`